Description: Properly inject network_data.json in configdrive
  The file "network_data.json" is not currently found in the folder
  "openstack/2015-10-15/" of config drive, only in "openstack/latest/".
  The patch makes sure its found in both folders.
Author: Mathieu GagneÃÅ <mgagne@iweb.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1513267
Origin: https://review.openstack.org/#/c/295982/

---

diff --git a/nova/api/metadata/base.py b/nova/api/metadata/base.py
index 9173c83..3d2355c 100644
--- a/nova/api/metadata/base.py
+++ b/nova/api/metadata/base.py
@@ -479,11 +479,11 @@
                 path = 'openstack/%s/%s' % (version, VD_JSON_NAME)
                 yield (path, self.lookup(path))
 
-        for (cid, content) in six.iteritems(self.content):
             if self._check_version(LIBERTY, version, ALL_OPENSTACK_VERSIONS):
                 path = 'openstack/%s/%s' % (version, NW_JSON_NAME)
                 yield (path, self.lookup(path))
 
+        for (cid, content) in six.iteritems(self.content):
             yield ('%s/%s/%s' % ("openstack", CONTENT_DIR, cid), content)
 
 
diff --git a/nova/tests/unit/test_metadata.py b/nova/tests/unit/test_metadata.py
index dc20731..8e72149 100644
--- a/nova/tests/unit/test_metadata.py
+++ b/nova/tests/unit/test_metadata.py
@@ -335,8 +335,32 @@
         fakes.stub_out_key_pair_funcs(self.stubs)
         inst = self.instance.obj_clone()
         inst_md = base.InstanceMetadata(inst)
+        expected_paths = [
+            'ec2/2009-04-04/user-data',
+            'ec2/2009-04-04/meta-data.json',
+            'ec2/latest/user-data',
+            'ec2/latest/meta-data.json',
+            'openstack/2012-08-10/meta_data.json',
+            'openstack/2012-08-10/user_data',
+            'openstack/2013-04-04/meta_data.json',
+            'openstack/2013-04-04/user_data',
+            'openstack/2013-10-17/meta_data.json',
+            'openstack/2013-10-17/user_data',
+            'openstack/2013-10-17/vendor_data.json',
+            'openstack/2015-10-15/meta_data.json',
+            'openstack/2015-10-15/user_data',
+            'openstack/2015-10-15/vendor_data.json',
+            'openstack/2015-10-15/network_data.json',
+            'openstack/latest/meta_data.json',
+            'openstack/latest/user_data',
+            'openstack/latest/vendor_data.json',
+            'openstack/latest/network_data.json',
+        ]
+        actual_paths = []
         for (path, value) in inst_md.metadata_for_config_drive():
+            actual_paths.append(path)
             self.assertIsNotNone(path)
+        self.assertEqual(expected_paths, actual_paths)
 
     def test_InstanceMetadata_queries_network_API_when_needed(self):
         network_info_from_api = []
